aikau/src/main/resources/alfresco/services/FileUploadService.js

```diff
@@ -0,0 +1,708 @@
+       * @instance
+       */
+      onUploadsContainerClosed: function alfresco_services_FileUploadService__onUploadsContainerClosed() {
+         this.reset();
+      },
+
+      /**
+       * Called when an upload "load" event fires. This merely means that the
+       * server has responded though, so it could still be an error.
+       *
+       * @instance
+       * @param {object} fileId The unique identifier of the file
+       * @param {object} evt The completion event
+       */
+      processUploadCompletion: function alfresco_services_FileUploadService__processUploadCompletion(fileId, evt) {
+
+         // Check the response code
+         var fileInfo = this.fileStore[fileId],
+            responseCode = fileInfo.request.status,
+            successful = responseCode >= 200 && responseCode < 300;
+
+         // Handle according to success
+         if (successful) {
+
+            // Get the response and update the file-info object
+            var response = JSON.parse(fileInfo.request.responseText);
+            fileInfo.nodeRef = response.nodeRef;
+            fileInfo.fileName = response.fileName;
+            fileInfo.state = this.state.SUCCESS;
+
+            // Notify uploads-display widget of completion
+            this.uploadDisplayWidget.handleCompletedUpload(fileId, evt, fileInfo.request);
+
+            // Execute post-upload actions
+            this.onUploadFinished();
+
+         } else {
+            this.processUploadFailure(fileId, evt);
+         }
+      },
+
+      /**
+       * Called if a request fails or completes with a non-success status code.
+       *
+       * @instance 
+       * @param {object} fileId The unique identifier of the file
+       * @param {object} evt The completion event
+       */
+      processUploadFailure: function alfresco_services_FileUploadService__processUploadFailure(fileId, evt) {
+         var fileInfo = this.fileStore[fileId];
+         if (fileInfo) {
+            fileInfo.state = this.state.FAILURE;
+            this.uploadDisplayWidget.handleFailedUpload(fileId, evt, fileInfo.request);
+            this.onUploadFinished();
+         }
+      },
+
+      /**
+       * Register this service's subscriptions.
+       * 
+       * @instance
+       * @override
+       * @listens module:alfresco/core/topics#UPLOAD_COMPLETION_ACKNOWLEDGEMENT
+       * @listens module:alfresco/core/topics#UPLOAD_CANCELLATION
+       */
+      registerSubscriptions: function alfresco_services_FileUploadService__registerSubscriptions() {
+         this.alfSubscribe(this.uploadTopic, lang.hitch(this, this.onUploadRequest));
+         this.alfSubscribe(topics.STICKY_PANEL_CLOSED, lang.hitch(this, this.onUploadsContainerClosed));
+      },
+
+      /**
+       * <p>Reset the state of the service.</p>
+       *
+       * <p><strong>NOTE:</strong> This does not cancel any in-progress uploads.</p>
+       *
+       * @instance
+       */
+      reset: function alfresco_services_FileUploadService__reset() {
```

Comment:
Suitable for mixin - this module would have the default, the UploadService would extend to perform the additional actions